src/System.CommandLine/Completions/CompletionContext.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.CommandLine.Parsing;
+
+namespace System.CommandLine.Completions
+{
+    /// <summary>
+    /// Supports command line completion operations.
+    /// </summary>
+    public class CompletionContext
+    {
+        internal CompletionContext(ParseResult parseResult)
+        {
+            ParseResult = parseResult;
+            TextToMatch = parseResult.TextToMatch();
+            RawInput = parseResult.RawInput;
+        }
+
+        internal CompletionContext(
+            ParseResult parseResult,
+            int position)
+        {
+            ParseResult = parseResult;
+            TextToMatch = parseResult.TextToMatch(position);
+            Position = position;
```

Comment:
minor thing: `rawInput` and `position` still remain separated. Should `rawInput` be moved to `CompletionContext` constructor?

Reply:
Yes, I think we do want to move that in there and make it public. `rawText` could use a better name, too. It's been internal so far so I didn't worry as much about it.

Right now when position isn't specified we assume it's at the end of the line, so effectively `rawText.Length + 1`.

Reply:
> Should rawInput be moved to CompletionContext constructor? 

Right now, `CompletionContext.RawInput` is populated by the `ParseResult` but only if the `ParseResult` was generated using a complete command line string input, so it doesn't strictly need to be provided in the constructor, and having it there would allow someone to pass in a `rawText` that didn't match the one in the `ParseResult`. 

> what is the behavior when rawInput is not specified but position is?

This shouldn't be allowed, I think. If you don't have a `rawInput`, the `position` is meaningless. I'm looking into enforcing this through type differentiation, e.g. something like `CompletionContext` vs `PositionalCompletionContext`. Only the latter would expose a way to get completions using a position.